server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCreateAction.java

```diff
@@ -0,0 +1,158 @@
+public class QProfileCreateAction implements BaseQProfileWsAction {
+
+  private static final String PARAM_PROFILE_NAME = "name";
+  private static final String PARAM_LANGUAGE = "language";
+  private static final String PARAM_BACKUP_FORMAT = "backup_%s";
+
+  private final DbClient dbClient;
+
+  private final QProfileFactory profileFactory;
+
+  private final QProfileExporters exporters;
+
+  private final Languages languages;
+
+  private final ProfileImporter[] importers;
+
+  public QProfileCreateAction(DbClient dbClient, QProfileFactory profileFactory, QProfileExporters exporters, Languages languages, ProfileImporter[] importers) {
+    this.dbClient = dbClient;
+    this.profileFactory = profileFactory;
+    this.exporters = exporters;
+    this.languages = languages;
+    this.importers = importers;
+  }
+
+  public QProfileCreateAction(DbClient dbClient, QProfileFactory profileFactory, QProfileExporters exporters, Languages languages) {
+    this(dbClient, profileFactory, exporters, languages, new ProfileImporter[0]);
+  }
+
+  @Override
+  public void define(WebService.NewController controller) {
+    NewAction setDefault = controller.createAction("create")
+      .setSince("5.2")
+      .setDescription("Create a quality profile.")
+      .setPost(true)
+      .setResponseExample(getClass().getResource("example-create.json"))
+      .setHandler(this);
+
+    setDefault.createParam(PARAM_PROFILE_NAME)
+      .setDescription("The name for the new quality profile.")
+      .setExampleValue("My Sonar way")
+      .setRequired(true);
+
+    setDefault.createParam(PARAM_LANGUAGE)
+      .setDescription("The language for the quality profile.")
+      .setExampleValue("js")
+      .setPossibleValues(LanguageParamUtils.getLanguageKeys(languages))
+      .setRequired(true);
+
+    if (importers.length > 0) {
+      for (ProfileImporter importer : importers) {
+        setDefault.createParam(getBackupParamName(importer.getKey()))
+          .setDescription("A configuration file for " + importer.getName() + ".");
+      }
+    }
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    UserSession.get().checkLoggedIn().checkGlobalPermission(GlobalPermissions.QUALITY_PROFILE_ADMIN);
+
+    String name = request.mandatoryParam(PARAM_PROFILE_NAME);
+    String language = request.mandatoryParam(PARAM_LANGUAGE);
+
+    DbSession dbSession = dbClient.openSession(false);
+
+    try {
+      QProfileResult result = new QProfileResult();
+      QualityProfileDto profile = profileFactory.create(dbSession, QProfileName.createFor(language, name));
+      result.setProfile(profile);
+      if (importers.length > 0) {
+        for (ProfileImporter importer : importers) {
+          InputStream contentToImport = request.paramAsInputStream(getBackupParamName(importer.getKey()));
+          if (contentToImport != null) {
+            result.add(exporters.importXml(profile, importer.getKey(), contentToImport, dbSession));
```

Comment:
There is a refactoring ticket for this: http://jira.codehaus.org/browse/SONAR-6396 (I only reused existing component)